modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalRequest.java

```diff
@@ -64,16 +69,53 @@ public void setRankEvalSpec(RankEvalSpec task) {
         this.rankingEvaluationSpec = task;
     }
 
+    /**
+     * Sets the indices the search will be executed on.
+     */
+    public RankEvalRequest setIndices(String... indices) {
+        Objects.requireNonNull(indices, "indices must not be null");
+        for (String index : indices) {
+            Objects.requireNonNull(index, "index must not be null");
+        }
+        this.indices = indices;
+        return this;
+    }
+
+    /**
+     * @return the indices for this request
+     */
+    public String[] getIndices() {
+        return indices;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         rankingEvaluationSpec = new RankEvalSpec(in);
-
+        if (in.getVersion().onOrAfter(Version.V_6_3_0)) {
+            indices = in.readStringArray();
+        } else {
+            // older 6.2 format
```

Comment:
add to the comment readVint vs readInt just to clarify something that may not be clear at first?

Reply:
I don't know if this clarifies things a lot since it requires looking into what readStringArray does? At this point I wonder if I should remove the comment completely, the code should explain this by itself and I think we can remove it on master after 6.3 is release (we only try to be bwc with the latest minor release before a major, no?)

Reply:
I think the comment may be useful, to clarify why we have this conditional as it looks weird at first. Of course if one looks deeper will also find out why we have it. Up to you